### ISSUE TITLE
Firebase XCFrameworks are not codesigned

### ISSUE THREAD

**Issue by jmagman** [firebase/firebase-ios-sdk#12238]
### Description

Third-party SDK XCFrameworks now need to be codesigned, not just the framework binaries within them.  This will be [enforced by Xcode at some point in Spring 2024](https://developer.apple.com/news/?id=r1henawx).

```
codesign --sign "Apple Developer cert etc" ... path/to/FirebaseAuth.xcframework
```

> Now with signatures for SDKs, when you adopt a new version of a third-party SDK in your app, Xcode will validate that it was signed by the same developer, improving the integrity of your software supply chain.

https://developer.apple.com/support/third-party-SDK-requirements/

Note `xcframework` is a directory (bundle), not a binary.  Though looks like Firebase binaries aren't codesigned either.

### Docs
https://developer.apple.com/documentation/xcode/verifying-the-origin-of-your-xcframeworks
https://developer.apple.com/videos/play/wwdc2023/10060/
![Screenshot 2024-01-03 at 4 35 03 PM](https://github.com/flutter/flutter/assets/682784/bf567f8f-a38f-43ff-a2db-d32c7e5b163c)

cc @paulb777 
(We're working on this for Flutter https://github.com/flutter/flutter/issues/140934)

### Reproducing the issue

Download XCFrameworks from https://firebase.google.com/docs/ios/setup#frameworks.  Drag into Xcode 15+, see `Kind: Unsigned` in the inspector.

![Screenshot 2024-01-03 at 4 36 25 PM](https://github.com/flutter/flutter/assets/682784/0c12028c-d5a8-4c64-9a3e-1a40549ff9f4)

```
$ codesign -d -vv Firebase/FirebaseAuth/FirebaseAuth.xcframework
Firebase/FirebaseAuth/FirebaseAuth.xcframework: code object is not signed at all
```

### Firebase SDK Version

10.19

### Xcode Version

15

### Installation Method

Zip

### Firebase Product(s)

All

### Targeted Platforms

iOS

**Comment by paulb777** [comment]
@jmagman Thanks for the report. @ncooke3 is in progress on meeting the new Apple signing requirements along with the privacy manifests  #11490

**Comment by jmagman** [comment]
👍  I saw the privacy manifest work but I didn't see anything about the signature requirement.  Thanks!

**Comment by hendri-voodoo** [comment]
hello, any ETA when this codesigning will be included?

**Comment by hendri-voodoo** [comment]
@ncooke3 noted. When is your next release cycle?

**Comment by pavm035** [comment]
Hi Any updates on signing?

**Comment by lakshmankreditbee** [comment]
Hi Any updates on signing?

**Comment by GH-Ong** [comment]
Hi is it possible to provide an ETA for release with code signing?

**Comment by vksgautam1986** [comment]
Hi Team any timeline where we can look for this ?

**Comment by paulb777** [comment]
The target window is between late March and mid April and we're working to do it as soon as we can in that timespan.

**Comment by nicolobozzato** [comment]
Hi! I saw release 10.23.0 signed xcframeworks for Firestore. Sorry to ask but what is the ETA for the other packages?
I'm sure it's already known but all the Firebase sdk are part of the special group that will get scrutiny by Apple
https://developer.apple.com/support/third-party-SDK-requirements/

**Comment by paulb777** [comment]
We're working towards providing signed xcframeworks in the next minor release, due out the week of April 8th

**Comment by shingt** [comment]
> We're working towards providing signed xcframeworks in the next minor release, due out the week of April 8th

Will the non-Firebase xcframeworks such as GoogleSignIn be also a target of the next release?
For example, GoogleSignIn supports the privacy manifest in [7.1.0 release](https://github.com/google/GoogleSignIn-iOS/releases/tag/7.1.0). I expect the new version of the xcframework is bundled in the Firebase release and it is codesigned, but am not sure if it is tracked.

**Comment by ncooke3** [comment]
@shingt, yes. Every framework in the Firebase.zip will have a code signature.

**Comment by wojciech-kulik** [comment]
Which release will have this fix? 10.23.2?

**Comment by ncooke3** [comment]
10.24.0, which is scheduled to be released next week!

**Comment by iOSNinja** [comment]
Any update on the signed SDKs release yet?

**Comment by wojciech-kulik** [comment]
@iOSNinja just a hint: you can use 10.23.0 with Xcode 14.2 as a workaround. I just sent the app to the review and it passed the validation.

**Comment by iOSNinja** [comment]
Thanks for the workaround @wojciech-kulik but i've upgraded all our build machines to > Xcode15 and was waiting for firebase to release the signed SDKs today as mentioned above.

**Comment by paulb777** [comment]
The 10.24.0 zip release is now available.

**Comment by techbyte24** [comment]
Has the Unity Firebase SDK provided signed xcframeworks for this points to version[10.24.0]( https://firebase.google.com/support/release-notes/ios#version_10240_-_april_9_2024 ) of the iOS SDK which provided signed yet? Which version is it?

**Comment by ncooke3** [comment]
This issue was fixed and released in 10.24.0. The Firebase Unity SDK was updated to use 10.24.0 in v11.9.0 (https://github.com/firebase/firebase-unity-sdk/issues/991).

There is now a newer Firebase version to use, so I recommend upgrading to the latest Firebase version (currently 10.25.0) as that included some additional fixes.
